### DISCUSSION TITLE
Integrazione tapparelle su home Assistant

### DISCUSSION THREAD

**Discussion by Batteralex** [indomus/forum#1031] [GENERALE - Discussioni generiche]
Salve, riporto qui la mia problematica riguardo la guida di integrazione tapparelle su home Assistant seguita con la configurazione della cover. Le tapparelle sono comandate da un sonoff 4ch pro e già qui mi chiedo come facciano i comandi a capire quale canale devono usare in base alla tapparella che sto comandando.premetto che i sonoff che uso in tutta la casa sono già configurati con mqtt e funzionano a meraviglia ma per le tapparelle usavo un pulsante fisico che bypassava lo stato del sonoff stesso quindi a fine ricablaggio corretto ho aggiunto  la configurazione cover ma purtroppo non funziona quindi chiedo il vostro aiuto per risolvere. Grazie anticipatamente. Alex

**Comment by snakuzzo** [comment]
Non ho capito la domanda sul "come facciano i comandi a capire quale canale usare"...sono cose che configuri tu lato Tasmota.
Scusa, ma se bypassi il sonoff...il sonoff non saprà lo stato reale della tapparella, o sbaglio?

**Reply by snakuzzo** [reply]
ad ogni modo... posta tutta la configurazione che hai fatto lato HA e cosa viene loggato sulla console tasmota quando provi a pilotare la tapparella da GUI Tasmota...perché spero che almeno da lì funzioni, altrimenti è tutto inutile

**Reply by Batteralex** [reply]
Si le tapparelle azionate direttamente da sonoff funzionano e per quanto riguarda il pulsante fisico e’ solamente un prolungamento del tasto posizionato sul sonoff stesso così che ad ogni azione il sonoff ne conosce lo stato. Per la configurazione da postare appena torno a casa la condivido in ogni caso ripeto che l’unica cosa che non funziona e’ la parte di configurazione “cover” presa dalla guida è incollata esattamente per com’era nel file cover.yaml ovviamente con il nome topic cambiato

**Reply by snakuzzo** [reply]
Sì ma tu postala lo stesso...e soprattutto devi postare quello che avviene in console Tasmota

**Reply by Batteralex** [reply]
Buonasera, come ti dicevo inizialmente quando ho cominciato ad usare HA molte integrazioni non c’erano e bisognava usare componenti aggiuntivi e configurazioni nel file yaml. adesso sto facendo un po’ di pulizia e mi ritrovo di fronte ad un dubbio: ad esempio il componente aggiuntivo mqtt potrebbe essere eliminato visto che mi sono ritrovato nelle integrazioni già configurate lo stesso mqtt? O sono due cose diverse ?

**Reply by snakuzzo** [reply]
te lo ritrovi configurato perché lo hai configurato nel file yaml. E non si tratta di alcun componente aggiuntivo.
Devi lasciare tutto così com'è

**Reply by Batteralex** [reply]
Ok. Quindi stessa cosa anche con HomeKit che si trova sia in config che in integrazione, inoltre ho notato molte entità chiamate sonoff tasmota, anche le stesse cover fatte ultimamente che anziché avere il nome datogli si chiamano tasmota. Come mai?

**Reply by snakuzzo** [reply]
Sì però ora stiamo andando off-topic. Se ti servono delucidazioni su questi argomenti leggi le guide o magari vieni in chat.
Ad ogni modo, il fatto che tu abbia in YAML la configurazione e poi te la ritrovi tra le integrazioni è del tutto normale (come ti ho già detto). E' la configurazione in YAML che te la fa vedere abilitata sotto integrazioni. Poi ci sono le integrazioni che non possono essere abilitate solo da interfaccia grafica (e quindi solo sotto integrazioni) e non da YAML.

Per quanto riguarda quelle entità Tasmota di cui parli, immagino tu ti riferisca a quelle dell'integrazione Tasmota.
Oltre alla configurazione manuale via YAML con platform MQTT, i device tasmota possono essere integrati con autodiscovery  in due modi:

1. autodiscovery (integrazione Tasmota). Si abilita sui device impostando `SO19 0` (che è quello di default)
2. autodiscovery MQTT. Si abilita sui device impostando `SO19 1`
 
Però ripeto...qui ora stiamo parlando di tutt'altra cosa. Quindi ti inviterei a chiudere questa specifica discussione. Grazie

**Reply by Batteralex** [reply]
Si scusami. Hai ragione.

**Comment by Batteralex** [comment]
allora di seguito ciò che mi appare in console tasmota non appena provo ad abbassare o alzare la tapparella:
14:34:34.325 MQT: stat/serranda_camerette/RESULT = {"Command":"Unknown"}
14:34:39.069 MQT: stat/serranda_camerette/RESULT = {"Command":"Unknown"}

mentre la configurazione nel file cover.yaml che ho copiato dalla guida e inserito è la seguente:
[Untitled-1.md](https://github.com/indomus/forum/files/7527933/Untitled-1.md)

**Reply by snakuzzo** [reply]
rispondi sotto l'ultimo messaggio...non creando nuovi thread.
Comunque si vede che stai facendo una publish errata

**Reply by Batteralex** [reply]
scusami uso pochissimo i forum quindi molte accortezze mi sfuggono.

**Reply by snakuzzo** [reply]
e poi...tu ti sei basato sulla configurazione di un'entità basata su un Sonoff Dual R2.
Sul 4CH dovrai adeguare NON SOLO il nome...ma sicuramente anche i command topic e quelli di telemetria. Non è un copia-incolla-sostituisci

**Reply by Batteralex** [reply]
certo infatti nella mia domanda iniziale postata appena sotto la guida avevo specificato di avere un 4ch e come quella configurazione potesse riuscire a comandare il canale giusto.
In ogni caso detto questo quali modifiche dovrei apportare alla configurazione per poterla fare funzionare con il 4ch?

**Reply by snakuzzo** [reply]
devi vedere com'è configurato lato Tasmota...ci devi arrivare tu... non posso dirtelo io al volo per farti fare un copia-incolla.
Leggi la documentazione Tasmota per capire innanzi tutto se hai configurato tutto per bene per un 4CH -> [QUI](https://tasmota.github.io/docs/Blinds-and-Shutters/)
Poi di sicuro dovrai cambiare sia i topic e payload per dare i comandi, ma questo potrai saperlo dopo aver letto la documentazione appena linkata.
Stessa cosa dicasi per i position topic (in riferimento all'errore che hai indicato prima).
Poi c'è da considerare se ci sono stati o meno cambiamenti lato Tasmota nel tempo

**Reply by snakuzzo** [reply]
Comunque ti ripeto...posta la configurazione delle DUE entità (le due tapparelle) che hai creato

**Reply by Batteralex** [reply]
ok grazie per i link che guarderò immediatamente.
nel frattempo ti posto la configurazione delle due tapparelle gestite dallo stesso sonoff:
[tapparelle.md](https://github.com/indomus/forum/files/7528159/tapparelle.md)

**Reply by snakuzzo** [reply]
senza nemmeno indagare più di tanto...ti dico già che è tutto sbagliato...e se non hai l'interlock è pure pericoloso.
Hai creato 4 switch per gestire 2 tapparelle...e vedo pure che non hai assolutamente seguito la logica che indicavamo in guida.
Ti consiglio di partire da zero e rifare tutto dopo aver letto la documentazione e riletto la guida.
Devi creare due entità `cover` e non 4 switch

**Reply by Batteralex** [reply]
ma considera che questa configurazione è stata fatta anni fa quando ancora molte integrazioni non esistevano e molte cose erano un po diverse. da li alcune cose tipo  switch e light non le ho mai più cambiate andando avanti cosi. infatti inzialmente ti avevo anche detto che comandavo le tapparelle con un pulsante fisico interposto tra le uscite sonoff e la tapparella stessa quindi bypassando del tutto lo stato sonoff. cmq l'interlock ce l'ho, almeno quello ;-)

**Reply by snakuzzo** [reply]
Sì ma anche anni fa (almeno due) quando la funzionalità per la gestione delle tapparelle non era ancora ufficialmente su firmware Tasmota, ma solo sulla fork di stefanbode, si usavano le `cover`

**Reply by Batteralex** [reply]
si vero però avendo sin dall'inizio fatto l'impianto con i pulsanti bypassanti non mi sono mai messo a riconfigurare niente lasciando tutto cosi com'era. adesso invece vorrei perfezionare la domotica e il suo funzionamento quindi sto modificando tutti i cablaggi sonoff/pulsanti/motori e di conseguenza mi sto ritrovando a configurazioni obsolete come dicevi prima tu.
in ogni caso sto gia seguendo la guida che mi hai girato ma onestamente non riesco a programmare tutti i canali del sonoff inoltre dopo aver settato i secondi di apertura e chiusura di una tapparella praticamente all'avvio parte e si blocca subito mentre la barra continua a scorrere.

**Reply by snakuzzo** [reply]
su questo non saprei come aiutarti, mi spiace.

**Reply by Batteralex** [reply]
grazie per avermi girato la guida con la quale finalmente dopo tante peripezie sono riuscito a fare le giuste configurazioni. l'unica cosa che ho notato non funzionare è l'aggiornamento delle posizioni delle tapparelle in home assistant nonostante abbia inserito l'automatione che ne richiede lo stato costantemente. Sai suggerirmi dove poter andare a controllare e cosa potrebbe non funzionare correttamente.
Grazie in anticipo.

**Reply by snakuzzo** [reply]
Si vede che hai sbagliato il position topic...devi far vedere cosa manda su mqtt...senza evidenze è del tutto inutile dire "non funziona"

**Reply by Batteralex** [reply]
questo è ciò che si vede da console nel sonoff:

```
13:38:53.418 MQT: tele/serrande_stanze_letto_bagno_nott/STATE = {"Time":"2021-11-17T13:38:53","Uptime":"0T17:00:10","UptimeSec":61210,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"Alice-49836661","BSSId":"F4:92:BF:21:97:15","Channel":11,"Mode":"11n","RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:03"}}
13:38:53.424 MQT: tele/serrande_stanze_letto_bagno_nott/SENSOR = {"Time":"2021-11-17T13:38:53","Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":25,"Direction":0,"Target":25}}
```

mentre questa è la parte di configurazione nel file automation.yaml:

**announce**

```yaml
- alias: Tasmota Announce
  trigger:
  - event: start
    platform: homeassistant
  condition: []
  action:
  - service: mqtt.publish
    data:
      topic: cmnd/tasmotas/POWER
  - service: mqtt.publish
    data:
      topic: cmnd/tasmotas/SHUTTERPOSITION
      payload: ""
```

infine la parte di configurazione in cover.yaml:

**cover.md**

```yaml
- platform: mqtt
  name: "Studio"
  device_class: shade
  availability_topic: "tele/serranda_camerette/LWT"
  payload_available: "Online"
  payload_not_available: "Offline"
  command_topic: "cmnd/serranda_camerette/Backlog"
  payload_open: "ShutterOpen2"
  payload_close: "ShutterClose2"
  payload_stop: "ShutterStop2"
  set_position_topic: 'cmnd/serranda_camerette/ShutterPosition2'
  position_topic: "stat/serranda_camerette/RESULT"
  position_open: 100
  position_closed: 0
  position_template: >
    {% if ('Shutter2' in value_json) and ('Position' in value_json.Shutter2) %}
      {{ value_json.Shutter2.Position }}
    {% else %}
      {% if is_state('cover.Studio', 'unknown') %}
        50
      {% else %}
        {{ state_attr('cover.Studio','current_position') }}
      {% endif %}
    {% endif %}  
  retain: false
  optimistic: false
  qos: 1
```

**Reply by snakuzzo** [reply]
Vedi che hai sbagliato...
La parte di log di Tasmota fa riferimento al topic `serrande_stanze_letto_bagno_nott` (per le quali ti suggerirei di usare topic meno prolissi perché potrebbero creare problemi topic troppo lunghi), mentre la parte di configurazione (`cover.md`) è relativa a `serranda_camerette`.

Tu devi fare così...
- ti metti sulla console tasmota
- dai il comando di apertura (o chiusura) e aspetti che si fermi la tapparella
- torni qui e pubblichi tutto quello che ha scritto sulla console da quando hai azionato il pulsante a quando si è fermata

P.S. anziché fare upload dei file usa il markdown...sia per la parte di log che per il codice YAML (mi sono permesso di modificare il tuo topic...così se vai in edit vedi come si usa)

**Reply by Batteralex** [reply]
no scusami, ho solo sbagliato a copiare e incollare la configurazione cover relativa alla giusta serranda.
fondamentalmente è la stessa identica ma cambia il nome topic

P.S. hai fatto bene a modificare, in realtà non sapevo come usare il markdown ho direttamente usato visualstudio e salvato da li.

**Reply by snakuzzo** [reply]
ok... correggi un attimo il codice nel precedente post e posta la parte di Tasmota a cui mi riferivo nei punti del mio post

**Reply by Batteralex** [reply]
ecco qui il log dopo aver aperto la serranda:

```
14:08:50.751 MQT: stat/serranda_camerette/RESULT = {"POWER3":"ON"}
14:08:50.754 MQT: stat/serranda_camerette/POWER3 = ON
14:08:50.757 MQT: stat/serranda_camerette/RESULT = {"ShutterPosition2":100}
14:08:50.761 MQT: stat/serranda_camerette/RESULT = {"Shutter2":{"Position":22,"Direction":1,"Target":100}}
14:08:50.898 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":23,"Direction":1,"Target":100}}
14:08:51.916 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":28,"Direction":1,"Target":100}}
14:08:52.893 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":33,"Direction":1,"Target":100}}
14:08:53.895 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":38,"Direction":1,"Target":100}}
14:08:54.897 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":44,"Direction":1,"Target":100}}
14:08:55.921 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":49,"Direction":1,"Target":100}}
14:08:56.923 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":54,"Direction":1,"Target":100}}
14:08:57.898 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":59,"Direction":1,"Target":100}}
14:08:58.898 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":65,"Direction":1,"Target":100}}
14:08:59.898 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":70,"Direction":1,"Target":100}}
14:09:00.914 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":75,"Direction":1,"Target":100}}
14:09:01.914 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":81,"Direction":1,"Target":100}}
14:09:03.035 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":86,"Direction":1,"Target":100}}
14:09:03.897 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":91,"Direction":1,"Target":100}}
14:09:04.898 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25},"Shutter2":{"Position":96,"Direction":1,"Target":100}}
14:09:05.616 MQT: stat/serranda_camerette/RESULT = {"POWER3":"OFF"}
14:09:05.620 MQT: stat/serranda_camerette/POWER3 = OFF
14:09:06.125 MQT: stat/serranda_camerette/SHUTTER2 = 100
14:09:06.130 MQT: stat/serranda_camerette/RESULT = {"Shutter2":{"Position":100,"Direction":0,"Target":100}}
```

**Reply by snakuzzo** [reply]
Allora...
innanzi tutto l'automazione di announce, per come è scritta, viene eseguita solo una volta...appena effettuato lo start di Home Assistant. A tal proposito ti chiederei un altro dettaglio:

- Vai `Configurazione > Automazioni` ed esegui a mano l'automazione di announce

e posta quello che viene loggato sulla console Tasmota

Poi...se non mi sfugge qualcosa, il `position_topic` è coerente ed il template pure, quindi mi sembra proprio strano che non si aggiorni il valore della posizione sull'entità di tipo cover che hai su HA. Su questo ti chiederei di guardare cosa viene loggato sul log di HA nel momento in cui fai una prova come quella appena fatta

**Reply by Batteralex** [reply]
questo è il log di tasmota non appena eseguita l'automazione announce:

```
14:22:31.340 MQT: stat/serranda_camerette/RESULT = {"POWER1":"OFF"}
14:22:31.343 MQT: stat/serranda_camerette/POWER1 = OFF
14:22:31.404 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":65,"Direction":0,"Target":65}}
14:22:31.410 MQT: stat/serranda_camerette/RESULT = {"ShutterPosition1":65}
```

mentre nei registri di HA stranamente non vedo nessun log riferito a questo "argomento"

**Reply by snakuzzo** [reply]
Ok...da quello che vedo tu dovresti avere il problema SOLO allo start di HA quando viene eseguita quell'automazione.
Questo perché, come potrai notare Tasmota pubblica la sola posizione di `Shutter1` a valle del topic di announce.

Quindi ti direi di provare un attimo con questa automazione. Lanciala come hai fatto prima e fammi vedere cosa scrive sulla console per favore

```yaml
- alias: Tasmota Announce
  trigger:
  - event: start
    platform: homeassistant
  action:
  - service: mqtt.publish
    data:
      topic: cmnd/tasmotas/POWER
  - service: mqtt.publish
    data:
      topic: cmnd/tasmotas/SHUTTERPOSITION
      payload: ""
  - service: mqtt.publish
    data:
      topic: cmnd/tasmotas/SHUTTERPOSITION2
      payload: ""
```

**Reply by Batteralex** [reply]
fatto...

```
14:39:27.552 MQT: stat/serranda_camerette/RESULT = {"POWER1":"OFF"}
14:39:27.557 MQT: stat/serranda_camerette/POWER1 = OFF
14:39:27.605 MQT: stat/serranda_camerette/RESULT = {"Shutter1":{"Position":66,"Direction":0,"Target":66}}
14:39:27.611 MQT: stat/serranda_camerette/RESULT = {"ShutterPosition1":66}
14:39:27.653 MQT: stat/serranda_camerette/RESULT = {"Shutter2":{"Position":67,"Direction":0,"Target":67}}
14:39:27.659 MQT: stat/serranda_camerette/RESULT = {"ShutterPosition2":67}
```
Una cosa che ho notato durante queste prove è che muovendo lo slide della cover su HA la serranda per la prima volta va dal lato opposto a dove dovrebbe andare (ad esempio lo muovo in chiusura ed invece si apre) e dalla seconda volta in più sale e scende in maniera corretta

P.S. come mai mostra solo POWER1 essendo un 4 CH?

**Reply by snakuzzo** [reply]
Ora dovrebbe darti la posizione di entrambe le tapparelle allo start di HA ... ti torna ? (se mi rispondi no vuol dire che hai i fantasmi dentro casa :D )
Per quanto riguarda lo slider, hai invertito i tasti lato device/tasmota
Sul POWER1...ti mostra quello perché tu pubblichi nell'automazione il topic con POWER ... che equivale a POWER1 ... è lo stesso motivo per cui prima ti pubblicava solo la posizione della tapparella 1 e non della 2

POWER = POWER1
SHUTTERPOSITION = SHUTTERPOSITION2

**Reply by Batteralex** [reply]
allo start di HA si ma la domanda è un altra?. se l'automazione announce si avvia solo quando parte HA cosa tiene aggiornato HA per tutto il resto delle modifiche di posizione delle serrande durante il giorno?... 
in ogni caso per adesso la tapparella per la quale ti ho mostrato i log sembra rispettare le posizioni anche su HA ma l'altra (quella gestita dall'altro sonoff che fondamentalmente è identica no). sale e scende ma lo slide non si aggiorna.

**Reply by snakuzzo** [reply]
La posizione si basa sulla pubblicazione del `position_topic` da parte del device, cosa che viene scatenata non solo quando tu pubblichi il topic di announce, ma anche quando il device ferma la tapparella nella posizione desiderata...

Però qui stiamo parlando delle basi del funzionamento del protocollo MQTT. Abbiamo degli articoli in merito e ti consiglio vivamente di leggerli.

Quanto all'altra tapparella, devi seguire la stessa logica che abbiamo seguito per la tapparella che ora funziona

**Reply by Batteralex** [reply]
Eccomi, allora per quanto riguarda le tapparelle legate al secondo sonoff per le quali lo slide posizione non si aggiorna ho ricontrollato sia la configurazione cover sia log lato tasmota rifacendo le prove che mi hai suggerito per il primo sonoff ma risulta tutto corretto nonostante continua a non aggiornare lo slide di posizione. C’è qualche altra prova che potrei fare ?

**Reply by snakuzzo** [reply]
se fosse tutto corretto si aggiornerebbe...
ti ho dato tutte le indicazioni necessarie...ora ci devi smanettare tu.
In alternativa mi dai accesso da remoto una di queste sere e controllo io se vuoi

**Reply by Batteralex** [reply]
Avevi ragione, l’errore era un misero carattere sbagliato nel nome topic in configurazione cover…

**Reply by snakuzzo** [reply]
Bene :)
